Disable colors in the output

eunomie opened the pull request:
It's a proposal to implement #205 

Color package has been taken from skaffold as mentionned in the issue. I don't know how you want to handle the copyright notice, if it's needed to be updated or not.

Two things have been changed from the skaffold's version:
- the default color is set to `None`
- a `NoColor` boolean has been introduced and `ColoredOutput` extends `IsTerminal` to know if colors should be printed

Tests has been updated.

Dependencies has been updated to be able to add go-cmp dependency for the tests.

The ~~first~~ third commit of the PR allows to automatically disable colors in the output when the output is not a terminal (like a file).

The ~~second~~ fourth introduces a `--no-color` flag that allows to force to disable colors, even if the output is a terminal. It can help on some CI by example.

In both cases, logrus configuration hasn't been changed, maybe I should do it when `--no-color` is set, WDYT?

nkubala commented on pkg/color/formatter.go:
```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2019 The Skaffold Authors
```
this line should look something like `Copyright 2019 Google Inc. All rights reserved.`

nkubala commented on pkg/color/formatter.go:
```diff
@@ -0,0 +1,144 @@
+type Color int
+
+var (
+	// LightRed can format text to be displayed to the terminal in light red, using ANSI escape codes.
```
I think you can remove these comments. it's pretty clear what each of these colors is doing, and is mostly visual clutter :) maybe one comment above the start of the `var` section about how these are all ANSI escape codes and that's it.

nkubala commented on pkg/color/formatter.go:
```diff
@@ -0,0 +1,144 @@
+	Cyan = Color(36)
+	// White can format text to be displayed to the terminal in white, using ANSI escape codes.
+	White = Color(37)
+	// None uses ANSI escape codes to reset all formatting.
```
this comment can stay

nkubala commented on pkg/color/formatter.go:
```diff
@@ -0,0 +1,144 @@
+	// None uses ANSI escape codes to reset all formatting.
+	None = Color(0)
+
+	// Default default output color for output from Skaffold to the user
```
s/`Skaffold`/`container-structure-test`

nkubala commented on pkg/color/formatter_test.go:
```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2019 The Skaffold Authors
```
update this copyright line as well

nkubala commented on pkg/color/formatter_test.go:
```diff
@@ -0,0 +1,131 @@
+	"io"
+	"testing"
+
+	"github.com/GoogleContainerTools/skaffold/testutil"
```
hmm, we don't want to depend on skaffold here. can you copy the relevant methods from the `testutil` package into the container-structure-test codebase? they can live in another `testutil` package here.

nkubala commented on pkg/color/formatter_test.go:
```diff
@@ -0,0 +1,131 @@
+func compareText(t *testing.T, expected, actual string, expectedN int, actualN int, err error) {
+	t.Helper()
+	if err != nil {
+		t.Errorf("Did not expect error when formatting text but got %s", err)
```
nit: errors shouldn't be capitalized

nkubala commented on pkg/color/formatter_test.go:
```diff
@@ -0,0 +1,131 @@
+	ColoredOutput = func(io.Writer) bool { return true }
+
+	var b bytes.Buffer
+	n, err := Green.Fprint(&b, "It's not easy being")
```
:frog:

eunomie commented:
Thanks for the review @nkubala 
I think I've fixed the comments.
I've added one commit to update the dependencies, to be able to add needed dependency for new tests.
And the commit to add skaffold colors has been split to be separated from its usage.

eunomie commented:
Hi @nkubala 
WDYT about the updated PR? Is that good for you?
Thanks
